src/components/FilterOverlay.vue

```diff
@@ -0,0 +1,128 @@
+                </v-list-item-content>
+              </v-list-item>
+            </v-list>
+            <v-card>
+              <v-divider />
+              <v-card-text>
+                <div>Choose a random subset of edges</div>
+                <v-slider
+                  v-model="subsetAmount"
+                  :max="max"
+                  :min="min"
+                  step="10"
+                  :ticks="true"
+                  thumb-label
+                >
+                  <v-text-field
+                    v-model="subsetAmount"
+                    class="mt-0 pt-0"
+                    hide-details
+                    single-line
+                    type="number"
+                    style="width: 60px"
+                  />
+                </v-slider>
+              </v-card-text>
+              <v-divider />
+            </v-card>
+
+            <v-btn
+              class="white--text"
+              color="teal"
+              @click="filterNetwork"
+            >
+              Filter Network
+            </v-btn>
+          </v-card>
+        </v-overlay>
+      </v-row>
+    </v-app>
+  </div>
+</template>
+
+<script lang="ts">
+/* eslint-disable no-restricted-syntax */
+import Vue from 'vue';
+import Vuex from 'vuex';
+import api from '@/api';
+import {
+  Network, Link,
+} from '@/types';
+import store from '@/store';
+
+Vue.use(Vuex);
+
+export default Vue.extend({
+  data: () => ({
+    overlay: true,
+    zIndex: 0,
+    min: 10,
+    max: 300,
+    subsetAmount: 0,
+  }),
+
+  computed: {
+    nodeTable() { return store.state.nodeTableName; },
+    workspace() { return store.state.workspaceName; },
+  },
+
+  methods: {
+    filterNetwork() {
+      if (this.workspace === null) {
+        return;
+      }
+
+      this.overlay = false;
+      const aqlQuery = `FOR nodes in ${this.nodeTable} LIMIT ${this.subsetAmount} FOR v,e,p in 1..4 ANY nodes GRAPH '${this.workspace}' LIMIT ${this.subsetAmount} RETURN p`;
+
+      const newTablePromise = api.aql(this.workspace, aqlQuery);
+
+      newTablePromise.then((promise) => {
+        const aqlNetwork: Network = { nodes: [], edges: [] };
+        // check for duplicates
+        const nodeChecker: string[] = [];
+        for (const path of promise) {
+          for (const arr of path.vertices) {
+            if (!nodeChecker.includes(arr._key)) {
+              nodeChecker.push(arr._key);
+            }
+          }
+        }
+        // create network in json format
+        for (const path of promise) {
+          for (const arr of path.vertices) {
+            if (nodeChecker.includes(arr._key)) {
+              arr.id = arr._id;
+              aqlNetwork.nodes.push(arr);
+              nodeChecker.splice(nodeChecker.indexOf(arr._key), 1);
+            }
+          }
+          path.edges.map((arr: Link) => aqlNetwork.edges.push(arr));
+        }
+        // Update state with new network
+        store.commit.setNetwork(aqlNetwork);
+      });
```

Comment:
You'd asked for my help refactoring this. Here's what I came up with. I used a set to track the `_id`, instead of key since that relies on the table. If you had nodes from 2 different tables with the same key, you might have had problems, and a set makes the logic much cleaner.
```suggestion
      newTablePromise.then((promise) => {
        const aqlNetwork: Network = { nodes: [], edges: [] };

        // check for duplicates
        const nodeChecker: Set<string> = new Set();
        promise.forEach((path) => {
          path.vertices.forEach((node: Node) => {
            nodeChecker.add(node._id);
          });
        });

        // create network in json format
        promise.forEach((path) => {
          path.vertices.forEach((node: Node) => {
            if (nodeChecker.has(node._id)) {
              aqlNetwork.nodes.push(node);
              nodeChecker.delete(node._id);
            }
          });
          aqlNetwork.edges = path.edges;
        });

        // Update state with new network
        store.commit.setNetwork(aqlNetwork);
      });
```

Reply:
If you accept this change, you'll also need to mess with the imports to add `Node`

Reply:
A set is so clever! and yes this refactoring is much easier to follow -- thank you!

Reply:
Resolved but slightly modified. Since there can be multiple edges in a path I add a map. Closed with c7ca8ee84382529a376ec2ea73171a6c4c7c39f3